lib/wallet/http.js

```diff
@@ -1993,6 +1993,56 @@ class HTTP extends Server {
 
       res.json(200, { success: true });
     });
+
+    this.get('/walletbackup', async (req, res) => {
+      const backup = [];
+      const walletIds = await this.wdb.getWallets();
+
+      for (const walletId of walletIds) {
+        if (walletId === 'primary') {
+          continue;
+        }
+
+        const wallet = await this.wdb.get(walletId);
+        const walletBackup = await util.backupWallet(wallet);
+        backup.push(walletBackup);
+      }
+
+      const buffer = util.zipObject(backup);
+      const backupName = `walletBackup_${util.timeStamp()}.json.gzip`;
+      res.setHeader('Content-Disposition', `inline; filename="${backupName}"`);
+      res.buffer(200, buffer);
+    });
+
+    this.post('/walletrestore', async (req, res) => {
+      const backup = req.body.backup;
```

Comment:
Similar comment -- we won't want to send secure backup material over this interface. Only the full node instance itself would even have access to the backups anyway. We should pass in a file name and have this code read from the file (which I see you've implemented below! I think we should get rid of this endpoint then)

Reply:
removed the interface, fixed ...